### PR TITLE
Only include infra cfg (proxy) nodes of active parents in the cfg (proxy) application

### PR DESCRIPTION
**Pull request by hakonhall**

### CODE REVIEW

File: node-repository/src/main/java/com/yahoo/vespa/hosted/provision/provisioning/Preparer.java [hunk 1]
```diff
@@ -72,6 +72,15 @@ public Preparer(NodeRepository nodeRepository, FlagSource flagSource, Optional<H
             List<Node> accepted = groupPreparer.prepare(application, clusterGroup,
                                                         requestedNodes.fraction(wantedGroups), surplusNodes,
                                                         indices, wantedGroups);
+
+            if (requestedNodes.requireActiveParent()) {
```
Comment (mpolden) [inline node-repository/src/main/java/com/yahoo/vespa/hosted/provision/provisioning/Preparer.java:76]: Shouldn't this be inverted? If we _not_  require active parents, we can ignore non-active parents.

Comment (hakonhall) [reply node-repository/src/main/java/com/yahoo/vespa/hosted/provision/provisioning/Preparer.java:76]: Yes, hold on

Comment (hakonhall) [reply node-repository/src/main/java/com/yahoo/vespa/hosted/provision/provisioning/Preparer.java:76]: Wait, I don't think your statement makes sense

Comment (freva) [reply node-repository/src/main/java/com/yahoo/vespa/hosted/provision/provisioning/Preparer.java:76]: Something needs to be inverted here, either the name or the implementation + statement

Comment (hakonhall) [reply node-repository/src/main/java/com/yahoo/vespa/hosted/provision/provisioning/Preparer.java]: Changes the method name to `rejectNonActiveParents()`

**Review by mpolden — APPROVED** [review]

**Pull request merged**